packages/grid/x-data-grid/src/internals/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -227,28 +230,25 @@ export const useGridPrintExport = (
       doc.current!.body.removeChild(printWindow);
 
       // Revert grid to previous state
-      apiRef.current.setState((state) => ({
-        ...state,
-        ...previousGridState.current,
-      }));
 
-      apiRef.current.unstable_enableVirtualization();
+      // We can not simply use setState(...previousGridState.current) because of this verification
```

Comment:
Could we use `apiRef.current.restoreState` once #3816 is merged ?
It has a small wrapper that avoids to run several `forceUpdate`.

Reply:
In fact I think you can already use it